common/python/connectors/common/db_helper/work_order_lmdb_helper.py

```diff
@@ -0,0 +1,164 @@
+            requested.
+        Returns jrpc response as defined in EEA spec 6.1.2
+        """
+
+        # Work order is processed if it is in wo-response table
+        value = self.kv_helper.get("wo-responses", wo_id)
+        if value:
+            response = json.loads(value)
+            if 'result' in response:
+                return response['result']
+
+            # response without a result should have an error
+            err_code = response["error"]["code"]
+            err_msg = response["error"]["message"]
+            if err_code == EnclaveError.ENCLAVE_ERR_VALUE:
+                err_code = WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE
+            elif err_code == EnclaveError.ENCLAVE_ERR_UNKNOWN:
+                err_code = WorkOrderStatus.UNKNOWN_ERROR
+            else:
+                err_code = WorkOrderStatus.FAILED
+            raise JSONRPCDispatchException(err_code, err_msg)
+
+        if(self.kv_helper.get("wo-timestamps", wo_id) is not None):
+            # work order is yet to be processed
+            raise JSONRPCDispatchException(
+                WorkOrderStatus.PENDING,
+                "Work order result is yet to be updated")
+
+        # work order not in 'wo-timestamps' table
+        raise JSONRPCDispatchException(
+            WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE,
+            "Work order Id not found in the database. Hence invalid parameter")
+
+# ---------------------------------------------------------------------------------------------
+    def submit_wo(self, wo_id, input_json_str):
+        """
+        Function to submit and store a new work-order
+        Parameters:
+            - wo_id: id of work-order being submitted
+            - input_json_str: The actual work-order as received
+            from the requester
+        """
+
+        if(self.kv_helper.get("wo-timestamps", wo_id) is None):
+
+            # Create a new work order entry.
+            # Don't change the order of table updation.
+            # The order is important for clean up if the TCS is restarted in
+            # the middle.
+            epoch_time = str(time.time())
+
+            # Update the tables
+            self.kv_helper.set("wo-timestamps", wo_id, epoch_time)
+            self.kv_helper.set("wo-requests", wo_id, input_json_str)
+            self.kv_helper.set("wo-scheduled", wo_id, input_json_str)
+
+            raise JSONRPCDispatchException(
+                WorkOrderStatus.PENDING,
+                "Work order is computing. Please query for WorkOrderGetResult \
+                 to view the result")
+
+        # Workorder id already exists
+        raise JSONRPCDispatchException(
+            WorkOrderStatus.INVALID_PARAMETER_FORMAT_OR_VALUE,
+            "Work order id already exists in the database. \
+                Hence invalid parameter")
+
+    def clear_a_processed_wo(self):
+        """
+        Function that clears one processed work-order from
```

Comment:
Update the comment that it will remove the least recently added entry from processed table.

Reply:
Done